solutions/cf-1000C.mdx

```diff
@@ -0,0 +1,77 @@
+---
+id: cf-1000C
+source: CF
+title: Covered Points COunt
+author: Jesse Choe
+---
+
+[Official Editorial](http://codeforces.com/blog/entry/60288)
+
+
+# Solution (Coordinate Compression)
+
+Due to the large constraints on $l_i$ and $r_i$, it is impossible to apply brute force the number of intersecting segments for each point on the coordinate line. Nonetheless, we can compress the coordinates and "pretend" as if the coordinates were between $1$ and $n$.
+
+This allows us to apply prefix sums on each point in our compressed "number line" and retransform these compressed coordinates to the original endpoints $l_i$ and $r_i$. To compress the coordinates, we can either sort the coordinates or use a `HashMap`.
+
+**Time Complexity:** $\mathcal{O}(N\log N)$
+
+<LanguageSection>
+
+<CPPSection>
+
+### Implementation
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+using ll = long long;
+using vi = vector<int>;
+using pi = pair<int, int>;
+
+#define f first
+#define s second
+#define all(x) begin(x), end(x)
+
+int main(){
+	int n; cin >> n;
+	vector<pair<ll, ll>> a(n);
```

Comment:
can you do better than `a`? i gave up on getting people to rename `n`

Reply:
I doubt it :man_shrugging:

Reply:
ok i glanced at the problem
`segments` would be a lot better

Reply:
too lazy :stuck_out_tongue:

Reply:
u can fix urself if it bothers u :laughing: